qcore/validate_vm.py

```diff
@@ -15,219 +15,282 @@
+        # Test all files we can, so we get all the problems at once
+        if file_path.exists():
+            errors.extend(validate_vm_file(file_path, vm_size))
 
-    # 6: binary contents
-    if numpy:
-        mins = []
-        for file_name in vm.values():
-            mins.append(
-                np.min(
-                    np.fromfile(
-                        file_name,
-                        dtype="<f{}".format(SIZE_FLOAT),
-                        count=vm_params_dict[VMParams.nz.value]
-                        * vm_params_dict[VMParams.nx.value],
-                    )
-                )
-            )
+    if vel_crns_file.exists():
+        polygon = []
+        with open(vel_crns_file) as crns_fp:
+            next(crns_fp)
+            next(crns_fp)
+            for line in crns_fp:
+                lon, lat = map(float, line.split())
+                lon = lon % 360
+                polygon.append((lon, lat))
+        if srf is not None:
+            srf_bounds = get_bounds(srf)
+        else:
+            srf_bounds = None
+        errors.extend(validate_vm_bounds(polygon, srf_bounds))
 
-        # works even if min is np.nan
-        if not min(mins) > 0:
-            return False, "VM vs, vp or rho <= 0|nan found: {}".format(vm_dir)
-
-    # 7: contents of meta files
-    #    if meta_created:
-    #        # TODO: check individual file contents
-    #        # not as important, can be re-created based on vm_params.py
-    #        pass
-
-    # 8: Check VM within bounds -If DEM file is not present, fails the VM
-
-    min_lat = -53
-    max_lat = -28
-    min_lon = 160
-    max_lon = 185
-    vel_crns_file = os.path.join(vm_dir, 'VeloModCorners.txt')
-    polygon = []
-
-    with open(vel_crns_file) as crns_fp:
-        next(crns_fp)
-        next(crns_fp)
-        for line in crns_fp:
-            lon, lat = map(float, line.split())
-            lon = lon % 360
-            polygon.append((lon, lat))
-            if lon < min_lon or lon > max_lon or lat < min_lat or lat > max_lat:
-                return False, "VM extents not contained within NZVM DEM"
-
-    # 9: Check SRF is within bounds of the VM if it is given
-    if srf is not None:
-        srf_bounds = get_bounds(srf)
+    if errors:
+        return False, "\n".join(errors)
+    return True, ""
+
+
+def validate_vm_bounds(polygon, srf_bounds=None):
+    """
+    Validates the VM domain against the DEM and the srf bounds
+    :param polygon: A list of (lon, lat) tuples giving the corners of the VM
+    :param srf: A list of (lon, lat) tuples giving the corners of the VM. Not used if None
+    :return: A list of error messages resulting from this validation
+    """
+    errors = []
+
+    for lon, lat in polygon:
+        if lon < MIN_LON or lon > MAX_LON or lat < MIN_LAT or lat > MAX_LAT:
+            errors.append(f"VM extents not contained within NZVM DEM: {lon}, {lat}")
+    # Check SRF is within bounds of the VM if it is given
```

Comment:
Would be good to add a comment that this is checked twice.

It's probably a good idea to check it twice as there may be slight differences in the corners to the computed bounds? (is that right?)

Reply:
Checked twice, as in once in the python script, once in the slurm script?

Reply:
Once in the check_params once in the check_NZVM?

Reply:
So the expected corners are tested against the DEM and srf in check_params, while the actual corners are tested against the DEM in check_NVZM (In the current vm_params2vm the srf isn't checked, that's done later at the install step atm). I don't know too much about how NZVM works, but I do expect there to be minor differences between python calculated expected, and C calculated actual corners.

Reply:
I don't think it's necessary to add a comment that it gets checked twice? Who would the comment be for and why would it be useful to them?

Reply:
Oh yeah. I see how that's getting called now. 

Yeah, the comment itself wouldn't be useful. I was thinking from the perspective that it's clear to which tests are part of which file. Which then that becomes a Wiki documentation thing.
What do you think?

Reply:
So making a wiki page for vm validation? Sounds like a plan